rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/CatalogUrlBuilder.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.rm.client.nodesource.serialization;
+
+import org.ow2.proactive_grid_cloud_portal.rm.shared.RMConfig;
+
+import com.google.gwt.core.client.GWT;
+
+
+public class CatalogUrlBuilder {
+
+    private String catalogUrl;
+
+    public String getCatalogUrl() {
+        if (this.catalogUrl == null) {
+            String catalogUrlFromConfig = RMConfig.get().getCatalogUrl();
+            String defaultCatalogUrl = GWT.getHostPageBaseURL().replace("/rm/", "/") + "catalog";
```

Comment:
Replacement "/rm/" to "/" is not clear for me. Could you please comment it

Reply:
Ok, the built-in `getHostPageBaseURL` returns the current URL, in which you have `rm` because you are located on the RM portal right now. If we want to query the catalog, the correct URL to use does not contain the `rm` path.